lib/Platform/Intl/impl_icu/Collator.cpp

```diff
@@ -0,0 +1,592 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include "Collator.h"
+
+#include "Constants.h"
+#include "IntlUtils.h"
+#include "Locale.h"
+#include "LocaleBCP47Object.h"
+#include "LocaleResolver.h"
+#include "OptionHelpers.h"
+#include "hermes/Platform/Intl/BCP47Parser.h"
+
+#include <algorithm>
+#include <optional>
+
+namespace hermes {
+namespace platform_intl {
+namespace impl_icu {
+
+constexpr auto &collationTypeDefault = u"default";
+constexpr auto &collationTypeSearch = u"search";
+constexpr auto &collationTypeStandard = u"standard";
```

Comment:
We should use the same style as `Constants.h`

Reply:
Done in 3rd commit.